### PR TITLE
feat: add `type` `switch` for `CSettingsItem`

### PR DESCRIPTION
**Pull request by ItsPierre**

### CODE REVIEW

File: src/components/CSettingsItem.vue [hunk 1]
```diff
@@ -26,18 +26,37 @@
...
 <script setup lang="ts">
 import { mdiChevronRight } from "@mdi/js";
 import { RouteLocationRaw } from "vue-router";
 
+/**
+ * The managed value is a value which will be managed by the settings item itself.
+ *
+ * It will be changed accordingly for some types like the `switch`.
+ */
+const managedValue = defineModel<boolean>();
```
Comment (johannes-huther) [inline src/components/CSettingsItem.vue:45]: switchValue or rename to type to toggle and name it toggleValue?

File: src/components/CSettingsItem.vue [hunk 2]
```diff
@@ -26,18 +26,37 @@
         </v-row>
       </div>
     </div>
+
+    <template v-if="type !== 'default'" #append>
```
Comment (johannes-huther) [inline src/components/CSettingsItem.vue]:
```suggestion
    <template v-if="type === 'switch'" #append>
```

**Review by johannes-huther — CHANGES_REQUESTED** [review]

File: src/components/CSettingsItem.vue (hunk 1 above)
Comment (ItsPierre) [reply src/components/CSettingsItem.vue:45]: But this will not only be the value for the switch. If we add another type, this should be used again

**Comment by ItsPierre** [comment]
Review: https://github.com/Crystal-Creations-GbR/crystal-components/compare/6f84400879afb51021aa269e4e4ab4f4f204af9a..9de493e16723df8d77a5bdbefc52f08f4ed30383

File: src/components/CSettingsItem.vue (hunk 1 above)
Comment (johannes-huther) [reply src/components/CSettingsItem.vue:45]: If I remember correctly you plan to merge value and managedValue at some point anyways? Please create an issue for that, then this is fine

Comment (ItsPierre) [reply src/components/CSettingsItem.vue:45]: yeah, I will probably refactor this whole component to make it easier to extend more types. See #105

**Review by johannes-huther — APPROVED** [review]

**Pull request merged**